x-pack/plugins/task_manager/server/task_running/task_runner.ts

```diff
@@ -242,30 +243,38 @@ export class TaskManagerRunner implements TaskRunner {
       );
     }
     this.logger.debug(`Running task ${this}`);
+
+    const apmTrans = apm.startTransaction(this.taskType, 'taskManager run', {
+      childOf: this.instance.task.traceparent,
+    });
+
     const modifiedContext = await this.beforeRun({
       taskInstance: this.instance.task,
     });
 
     const stopTaskTimer = startTaskTimer();
-    const apmTrans = apm.startTransaction(`taskManager run`, 'taskManager');
-    apmTrans?.addLabels({
-      taskType: this.taskType,
-    });
+
     try {
       this.task = this.definition.createTaskRunner(modifiedContext);
-      const result = await this.task.run();
+      const result = await withSpan({ name: 'run', type: 'task manager' }, () => this.task!.run());
```

Comment:
Just curious, why is `this.task!` necessary in this PR, but not in master? (on the above line)